### PR TITLE
Fixed #32367 -- Fixed models.W042 is raised on inherited manually specified primary key.

### PR DESCRIPTION
**Pull request by hramezani**
https://code.djangoproject.com/ticket/32367

### CODE REVIEW

File: tests/check_framework/test_model_checks.py [hunk 1]
```diff
@@ -395,6 +395,15 @@ class Model(models.Model):
             ),
         ])
 
+    def test_inherited__pk(self):
```
Comment (uri-rodberg) [inline tests/check_framework/test_model_checks.py:398]: Why do you need double underline? Is it a typo?

Comment (hramezani) [reply tests/check_framework/test_model_checks.py]: Yes, it was a typo. I just fixed it.

File: django/db/models/options.py [hunk 2]
```diff
@@ -280,6 +280,10 @@ def _prepare(self, model):
                 if already_created:
                     field = already_created[0]
                 field.primary_key = True
+                try:
+                    field.auto_created = field.reverse_related_fields[0][0].auto_created
+                except ValueError:
+                    pass
```
Comment (charettes) [inline django/db/models/options.py:286]: I think this logic would mark `Children.parent_ptr.auto_created = True` when it should be `False`

```python
class Parent(models.Model):
    pass

class Children(Parent):
    parent_ptr = models.OneToOne(Parent, parent_link=True)
```

Comment (charettes) [reply django/db/models/options.py:286]: Should the check logic be adapted to deal with `parent_link` primary key differently instead?

Comment (hramezani) [reply django/db/models/options.py:286]: Yes, I think you are right. 
I tried to check `field.remote_field.parent_link`, but it was `True` for both your mentioned test and the test that I added to the patch.

I don't know how we can distinguish between these two cases.

Comment (felixxm) [reply django/db/models/options.py:286]: I think we should ignore inherited PKs and check them only in parents, see #13925.

**Comment by felixxm** [comment]
Superseded by #13925.

**Pull request closed without merging**